### PR TITLE
fix a bunch of clippy warnings

### PR DESCRIPTION
**Pull request by matthiaskrgr**
clippy::bind_instead_of_map
clippy::branches_sharing_code
clippy::collapsible_match
clippy::inconsistent_struct_constructor
clippy::int_plus_one
clippy::iter_count
clippy::iter_nth_zero
clippy::manual_range_contains
clippy::match_like_matches_macro
clippy::needless::collect
clippy::needless_question_mark
clippy::needless_return
clippy::op_ref
clippy::option_as_ref_deref
clippy::ptr_arg
clippy::redundant_clone
clippy::redundant_closure
clippy::redundant_static_lifetimes
clippy::search_is_some
clippy::#single_char_add_str
clippy::single_char_pattern
clippy::single_component_path_imports
clippy::single_match
clippy::skip_while_next
clippy::unnecessary_lazy_evaluations
clippy::unnecessary_unwrap
clippy::useless_conversion
clippy::useless_format

### CODE REVIEW

File: src/formatting.rs [hunk 1]
```diff
@@ -266,14 +266,14 @@ impl FormattingError {
...
-            | ErrorKind::IoError(_)
-            | ErrorKind::ParseError
-            | ErrorKind::LostComment => true,
-            _ => false,
-        }
+                | ErrorKind::TrailingWhitespace
+                | ErrorKind::IoError(_)
+                | ErrorKind::ParseError
+                | ErrorKind::LostComment
+        )
```
Comment (calebcartwright) [inline src/formatting.rs:276]: This one as well

File: src/macros.rs [hunk 2]
```diff
@@ -690,25 +687,22 @@ fn delim_token_to_str(
...
     fn starts_with_brace(&self) -> bool {
-        match *self {
+        matches!(
+            *self,
             MacroArgKind::Repeat(DelimToken::Brace, _, _, _)
-            | MacroArgKind::Delimited(DelimToken::Brace, _) => true,
-            _ => false,
-        }
+                | MacroArgKind::Delimited(DelimToken::Brace, _)
+        )
```
Comment (calebcartwright) [inline src/macros.rs:694]: and revert this one

File: src/macros.rs [hunk 3]
```diff
@@ -1137,35 +1131,35 @@ enum SpaceState {
...
+            | TokenKind::Tilde
+            | TokenKind::BinOpEq(_)
+            | TokenKind::At
+            | TokenKind::RArrow
+            | TokenKind::LArrow
+            | TokenKind::FatArrow
+            | TokenKind::BinOp(_)
+            | TokenKind::Pound
+            | TokenKind::Dollar
+    )
```
Comment (calebcartwright) [inline src/macros.rs:1155]: and this one

**Review by calebcartwright — CHANGES_REQUESTED** [review]
Thank you for this and apologies for the delayed review. The vast majority of it looks good, but I don't want to merge as-is because some of the `matches!` changes with the larger/multiline pattern args don't provide us any value and actually have a negative hit on readability due to the extra indentation levels.

I realize a big part of my issue with these ties back to the fact that we format `matches` calls like any other macro, but that's going to continue to be the status quo for a while.

I've left inline notes on the ones that we'd need to have reverted back, and this will need a rebase as well

**Comment by matthiaskrgr** [comment]
Hope that was it, there was only a single merge conflict :)

**Comment by calebcartwright** [comment]
Awesome, thank you!

**Pull request merged**